core/src/main/java/io/questdb/griffin/engine/functions/NegatingFunctionFactory.java

```diff
@@ -59,6 +60,11 @@ public Function newInstance(
             SqlExecutionContext sqlExecutionContext
     ) throws SqlException {
         Function function = delegate.newInstance(position, args, argPositions, configuration, sqlExecutionContext);
+
+        // Return boolean constant if short/byte null comparison
+        if(byteRShortNullComparison(args)){
```

Comment:
Negated factory is not a great place to do this, it doesn't wrap all comparison operators. I feel that this comparison should by done on individual "Eq" factory level for each, byte and short

Reply:
Hmm, individual factories `EqShortFunctionFactory` and `EqByteFunctionFactory` still have the following comparison
```
if (left.isNullConstant() || right.isNullConstant()) {
            return BooleanConstant.FALSE;
        }
  ```  
Returning a BooleanConstant , the one in NegatingFactory is for `!=` comparison  
Not sure if I have misunderstood